### ISSUE TITLE
[Bug] 1-1 relationship field doesn't seem to work using class notation

### ISSUE THREAD

**Issue by mraap** [Laravel-Backpack/CRUD#4033]
# Bug report

### What I did

Added a 1-1 relationship field using dot-notation like so:
```
CRUD::field('rideWaitingTime.waiting_time')
    ->label('Wachttijd')
    ->type('number')
    ->attributes(['step' => 1])
    ->tab('Openings- en wachttijden');
```

### What I expected to happen

To see a number type field underneath the "Openings- en wachttijden" tab labeled "Wachttijd".

### What happened

A text field appeared on top of the form above the tabs. It was otherwise working properly, it loads existing data and stores it properly in the database, so the relationship works but somehow the other settings (label, type, attributes and tab) are being ignored.

### What I've already tried to fix it

I was able to fix it by declaring the field using $this->crud->addField, like this:
```
$this->crud->addField([
    'name' => 'rideWaitingTime.waiting_time',
    'label' => 'Wachttijd',
    'type' => 'number',
    'attributes' => [ 'step' => 1],
    'tab' => 'Openings- en wachttijden',
]);
```
This way it's working completely as expected, so it's not a huge problem, but it should probably work using the class notation too so I thought I'd mention the issue to you.

### Is it a bug in the latest version of Backpack?

After I run ```composer update backpack/crud``` the bug... is it still there?
Yes.

### Backpack, Laravel, PHP, DB version

When I run ```php artisan backpack:version``` the output is:
```
### PHP VERSION:
PHP 7.4.24 (cli) (built: Sep 24 2021 00:55:41) ( NTS )
Copyright (c) The PHP Group
Zend Engine v3.4.0, Copyright (c) Zend Technologies
    with Zend OPcache v7.4.24, Copyright (c), by Zend Technologies
    with Xdebug v3.0.4, Copyright (c) 2002-2021, by Derick Rethans
    with blackfire v1.67.0~linux-musl-x64-non_zts74, https://blackfire.io, by Blackfire

### LARAVEL VERSION:
v7.30.5@afb0c034072a03a5ab1872fbdea54f8befd873c3

### BACKPACK VERSION:
4.1.62@273c277e826c157ef0913962769a7ec9a44b6233
```
![fails](https://user-images.githubusercontent.com/7535679/147258906-99be2078-387b-4f30-a4ff-2d3e00621794.jpg)
![works](https://user-images.githubusercontent.com/7535679/147258918-90f86993-7c74-4dbf-a627-39857f30735b.jpg)

**Comment by pxpm** [comment]
@mraap thanks for the report. 

When I read your issue I was sure I already saw it before.

Here it is: https://github.com/Laravel-Backpack/CRUD/pull/3981

Make sure you are not overriding any files in your `app/resources` folder that were modified in that PR, if you do please manually add the fixes.

Let me know how it goes or feel free to close if I was rigth!

Merry Christmas,
Pedro

**Comment by pxpm** [comment]
Disregard my last comment @mraap . I was wrong you were rigth!

The fix had only been merged in 4.2. 

@tabacitu should we also merge that fix into 4.1 ? It's a little BC, some public methods removed, but still worth it for the use case ?

**Comment by tabacitu** [comment]
Thanks for reporting this @mraap ! It's indeed a bug, it's been already fixed in 4.2 so you'll be getting it very soon when we release 4.2 and you upgrade. Until then, please use `addField()` for that one... it will look odd but only for a while.

@pxpm - yup - that's what I wanted to say, it's exactly #3996 . No, we have no time to backport it to 4.1 right now, let's focus on finishing 4.2 please. Focus! 😅

**Comment by mraap** [comment]
I didn't find the other ticket about this issue so I just thought I'd mention it to you. It's no problem at all to use the addField for now. Thanks for clarifying 😊
